src/ppx_deriving_qcheck/ppx_deriving_qcheck.ml

```diff
@@ -253,43 +242,42 @@ let record ~loc ~gens ?(f = fun x -> x) xs =
   in
   let expr = A.pexp_record fields None |> f in
 
-  map ~loc pat expr gen
+  G.map ~loc pat expr gen
 
-let rec gen_from_type ~loc ?(env = TypeGen.empty) ?(typ_name = "") typ =
+(** {2. Core derivation} *)
+
+(** [gen_from_type typ] performs the AST traversal and derivation to qcheck generators *)
+let rec gen_from_type ~loc ~env typ =
   Option.value (Attributes.gen typ)
     ~default:
       (match typ with
-      | [%type: unit] -> [%expr QCheck.Gen.unit]
-      | [%type: int] -> [%expr QCheck.Gen.int]
-      | [%type: string] | [%type: String.t] -> [%expr QCheck.Gen.string]
-      | [%type: char] -> [%expr QCheck.Gen.char]
-      | [%type: bool] -> [%expr QCheck.Gen.bool]
-      | [%type: float] -> [%expr QCheck.Gen.float]
-      | [%type: int32] | [%type: Int32.t] -> [%expr QCheck.Gen.int32]
-      | [%type: int64] | [%type: Int64.t] -> [%expr QCheck.Gen.int64]
-      | [%type: [%t? typ] option] ->
-          [%expr QCheck.Gen.option [%e gen_from_type ~loc ~env typ]]
-      | [%type: [%t? typ] list] ->
-          [%expr QCheck.Gen.list [%e gen_from_type ~loc ~env typ]]
-      | [%type: [%t? typ] array] ->
-          [%expr QCheck.Gen.array [%e gen_from_type ~loc ~env typ]]
+      | [%type: unit] -> G.unit loc
+      | [%type: int] -> G.int loc
+      | [%type: string] | [%type: String.t] -> G.string loc
+      | [%type: char] -> G.char loc
+      | [%type: bool] -> G.bool loc
+      | [%type: float] -> G.float loc
+      | [%type: int32] | [%type: Int32.t] -> G.int32 loc
+      | [%type: int64] | [%type: Int64.t] -> G.int64 loc
+      | [%type: [%t? typ] option] -> G.option ~loc (gen_from_type ~loc ~env typ)
+      | [%type: [%t? typ] list] -> G.list ~loc (gen_from_type ~loc ~env typ)
+      | [%type: [%t? typ] array] -> G.array ~loc (gen_from_type ~loc ~env typ)
       | { ptyp_desc = Ptyp_tuple typs; _ } ->
           let tys = List.map (gen_from_type ~loc ~env) typs in
-          tuple ~loc tys
-      | { ptyp_desc = Ptyp_constr ({ txt = ty; _ }, _); _ } ->
-          let x = TypeGen.find_opt (longident_to_str ty) env in
-          Option.value ~default:(gen ~loc ~env ty) x
-      | { ptyp_desc = Ptyp_var s; _ } -> gen ~loc (Lident s)
+          gen_tuple ~loc tys
+      | { ptyp_desc = Ptyp_constr ({ txt = ty; _ }, args); _ } ->
+         let args = List.map (gen_from_type ~loc ~env) args in
+         gen_longident ~loc ~env ty args
+      | { ptyp_desc = Ptyp_var s; _ } -> gen_longident ~loc ~env (Lident s) []
```

Comment:
Insert  a newline, so that the `var` case body is consistent with `variant` and `arrow` below?

Reply:
Done, thank you